contribs/gnodev/pkg/dev/node_state.go

```diff
@@ -0,0 +1,140 @@
+package dev
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gnolang/gno/contribs/gnodev/pkg/events"
+	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
+
+// Save the current state as initialState
+func (n *Node) SaveCurrentState(ctx context.Context) error {
+	n.muNode.RLock()
+	defer n.muNode.RUnlock()
+
+	// Get current blockstore state
+	state, err := n.getState(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to save state: %s", err.Error())
+	}
+
+	n.initialState = state[:n.currentStateIndex]
+	return nil
+}
+
+// Export the current state as list of txs
+func (n *Node) ExportCurrentState(ctx context.Context) ([]std.Tx, error) {
+	n.muNode.RLock()
+	defer n.muNode.RUnlock()
+
+	// Get current blockstore state
+	state, err := n.getState(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to save state: %s", err.Error())
+	}
+
+	return state[:n.currentStateIndex], nil
+}
+
+func (n *Node) getState(ctx context.Context) ([]std.Tx, error) {
+	if n.state == nil {
+		var err error
+		n.state, err = n.getBlockStoreState(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("unable to save state: %s", err.Error())
+		}
+	}
+
+	return n.state, nil
+}
+
+// MoveFrom adjusts the current state of the node by `x` transactions.
+// `x` can be negative to move backward or positive to move forward, however, index boundaries are respected
+// with a lower limit of 0 and upper limit equaling the total number of states.
+// If a move is successful, node is reload.
+func (n *Node) MoveFrom(ctx context.Context, x int) error {
```

Comment:
Maybe a better name for this method would be `MoveBy`

Reply:
I struggle a lot with this name actually. I know something was wrong... ahahah. Thanks!